ludwig/config_validation/checks.py

```diff
@@ -291,7 +291,25 @@ def is_divisible(hidden_size: int, num_heads: int) -> bool:
             )
```

Comment:
While `hyperopt.parameters` remains a `Dict`, I might add a validation check here as well that loops through and checks basic stuff like:
1) is every nested parameter a dict itself
2) are the nested parameters valid keywords (e.g. is there something other than `space`, `lower`, `upper`, etc.)
3) if you want you could even try to dynamically apply the parameter space schemas here based on `space`

Reply:
Followup: Added checks for parameter spaces that instantiate the spaces as config objects. Additionally, there are basic checks to ensure that they refer to existing config parameters.